dependency/catalog_node.go

```diff
@@ -152,5 +154,5 @@ type NodeServiceList []*NodeService
 func (s NodeServiceList) Len() int      { return len(s) }
 func (s NodeServiceList) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
 func (s NodeServiceList) Less(i, j int) bool {
-	return s[i].Service <= s[j].Service
```

Comment:
I think we might want to order by service first, then fallback to ID. What do you think?